lib/post/widgets/report_post_dialog.dart

```diff
@@ -0,0 +1,209 @@
+import 'package:auto_size_text/auto_size_text.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+import 'package:thunder/feed/feed.dart';
+
+import 'package:thunder/post/bloc/post_bloc.dart';
+import 'package:thunder/shared/snackbar.dart';
+
+class ReportAPostDialog extends StatefulWidget {
+  const ReportAPostDialog({super.key, required this.postId, this.onReport});
+  final void Function(String)? onReport;
+  final int postId;
+  @override
+  State<ReportAPostDialog> createState() => _ReportAPostDialogState();
+}
+
+class _ReportAPostDialogState extends State<ReportAPostDialog> {
+  late TextEditingController messageController;
+  @override
+  void initState() {
+    messageController = TextEditingController();
+    super.initState();
+  }
+
+  @override
+  void dispose() {
+    messageController.dispose();
+    super.dispose();
+  }
+
+  bool hasError = false;
+  String errorMessage = '';
+  @override
+  Widget build(BuildContext context) {
+    return Material(
+      child: Padding(
+        padding: EdgeInsets.only(
+          bottom: MediaQuery.of(context).viewInsets.bottom,
+        ),
+        child: Padding(
+          padding: const EdgeInsets.symmetric(
+            horizontal: 18.0,
+            vertical: 12,
+          ),
+          child: Column(
+            mainAxisSize: MainAxisSize.min,
+            children: [
+              AutoSizeText(
+                AppLocalizations.of(context)!.reportAPost,
```

Comment:
For conventions sake, define `AppLocalizations.of(context)!` at the top of the build method and use `l10n.reportAPost`. The same goes for all the other instances of AppLocalizations.

```dart
Widget build(BuildContext context) {
  final l10n = AppLocalizations.of(context)!;

 ...
  AutoSizeText(l10n.reportAPost)
}
```